client/src/Interface.elm

```diff
@@ -0,0 +1,73 @@
+module Interface exposing (EvaluationResult(..), decodeEvaluationResult, parseEvaluationResult)
+
+import Json.Decode exposing (Decoder, andThen, string)
+import Parser exposing ((|.), (|=), DeadEnd, Parser, oneOf, run, succeed, symbol, variable)
```

Comment:
so elm has already a parser combinator? we can write the parser on the frontend for the languages as well ?